magento/pwa-studio
# [BUGFIX] Show a message when a category has no products assigned

## Description

This PR introduces a new component in `RootComponents/Category` called `NoProductsFound` which is rendered when a category has no products.

It also:
  * Includes a `NoProductsFound` talon in Peregrine
  * Removes `packages/peregrine/lib/store/actions/catalog/mockData.js` which wasn't being used and was out of date anyway (which caused confusion)

## Related Issue
Closes #823.
Closes #1678 

## Verification Steps

1. Create an empty category in M2 Admin by going to `Catalog` > `Categories` > `Add Subcategory`
1. Do not assign any products to your new category
1. In your browser's `Application` tab, click `Clear Site Data` and refresh the page
1. Navigate to Venia's homepage
1. See that your new empty category appears<sup>1</sup>
1. Click the empty category
1. See that you are presented with a friendly message and other categories are recommended to you<sup>2</sup>
1. Verify that the links work correctly (take you to the correct category page)

## Screenshots / Screen Captures (if appropriate)

<sup>1</sup>
![Screen Shot 2019-09-30 at 10 45 30 AM](https://user-images.githubusercontent.com/13182778/65896466-ea686d80-e372-11e9-9473-d2c80b36f70d.png)

<sup>2</sup>
![Screen Shot 2019-09-30 at 11 11 35 AM](https://user-images.githubusercontent.com/13182778/65896589-171c8500-e373-11e9-8d4c-35a286381a59.png)

## Proposed Labels for Change Type/Package




## Checklist:
